File: obsoleted/Tests/DataModelTests.m
// -*-  Mode:ObjC; c-basic-offset:4; tab-width:8; indent-tabs-mode:nil -*-

#import "TestCommon.h"

@interface DataModelTest : IUTTest {
    DataModel *dm;
}
@end


@implementation DataModelTest

- (void)setUp
{
    [TestCommon deleteDatabase];
    dm = [DataModel instance];
    [dm load];
}

- (void)tearDown
{
}

#pragma mark -
#pragma mark Tests

// データベースがないときに、初期化されること
- (void)testInitial
{
    // 初期データチェック
    Assert(dm != nil);
    AssertEqualInt(0, [dm.journal.entries count]);

    Asset *as = [dm.ledger.assets objectAtIndex:0];
    Assert([as.name isEqualToString:@"Cash"]); 
                  
    AssertEqualInt(0, [dm.categories count]);
}

// データベースがあるときに、正常に読み込めること
- (void)testNotInitial
{
    [TestCommon installDatabase:@"testdata1"];
    dm = [DataModel instance];

    Assert([dm.journal.entries count] == 6);
    Assert([dm.ledger.assets count] == 3);
    Assert([dm.categories count] == 3);
}

@end

File: CashFlowTests/LogicTests/DataModelTests.swift
import Foundation
import XCTest

@testable import CashFlow

class DataModelTest : XCTestCase {
    var dm : DataModel! = nil
    
    override func setUp() {
        super.setUp()
        TestCommon.initDatabase()
        dm = DataModel.instance()
    }

    override func tearDown() {
        super.tearDown()
    }

    // データベースがないときに、初期化されること
    func testInitial() {
        // 初期データチェック
        XCTAssert(dm != nil)
        XCTAssertEqual(0, dm.journal.entries.count)

        let asset = dm.ledger.assets[0]
        XCTAssertEqual(NSLocalizedString("Cash", comment:""), asset.name)
                  
        XCTAssertEqual(0, dm.categories.count)
    }

    // データベースがあるときに、正常に読み込めること
    func testNotInitial() {
        TestCommon.installDatabase("testdata1")
        dm = DataModel.instance()

        XCTAssertEqual(6, dm.journal.entries.count)
        XCTAssertEqual(3, dm.ledger.assets.count);
        XCTAssertEqual(3, dm.categories.count);
    }
}

